internal/httpapi/server/server.go

```diff
@@ -0,0 +1,91 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/sirupsen/logrus"
+
+	"github.com/redhat-data-and-ai/usernaut/internal/httpapi/middleware"
+	"github.com/redhat-data-and-ai/usernaut/pkg/config"
+)
+
+type APIServer struct {
+	config *config.AppConfig
+	router *gin.Engine
+	server *http.Server
+}
+
+func NewAPIServer(cfg *config.AppConfig) *APIServer {
+	if cfg.App.Environment == "local" {
+		gin.SetMode(gin.DebugMode)
+	} else {
+		gin.SetMode(gin.ReleaseMode)
+	}
+	router := gin.New()
+	router.Use(gin.Logger())
```

Comment:
instead of using `gin.Logger`, we should be using our logger. We can have this in a follow-up MR